src/transformStateWithClones.js

```diff
@@ -1,13 +1,31 @@
-'use strict';
-
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const stateHistory = [];
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(stateClone, obj.extraData);
+      stateHistory.push({ ...stateClone });
```

Comment:
[DONT REPEAT YOURSELF]: Instead of pushing the state clone to the history array within each case, do it once at the end of the loop after the switch statement to avoid code repetition.